File: api/src/test/java/net/blueshell/api/factory/dto/SocialDTOFactory.java
package net.blueshell.api.factory.dto;

import lombok.RequiredArgsConstructor;
import net.blueshell.api.common.enums.PlatformType;
import net.blueshell.api.dto.SocialDTO;
import org.springframework.stereotype.Component;

/**
 * Factory for SocialDTO test instances.
 */
@Component
@RequiredArgsConstructor
public class SocialDTOFactory extends BaseDtoFactory<SocialDTO> {

    @Override
    public Class<SocialDTO> targetType() {
        return SocialDTO.class;
    }

    @Override
    public SocialDTO createBasic() {
        SocialDTO dto = new SocialDTO();
        dto.setTitle("Hello world");
        dto.setText("Body");
        dto.setUrl("https://example.com/" + nextId());
        dto.setPlatforms(new PlatformType[]{PlatformType.FACEBOOK});
        return dto;
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/dto/SocialDTOFactory.kt
package net.blueshell.api.factory.dto

import net.blueshell.api.blog.web.dto.SocialDTO
import net.blueshell.api.shared.enums.PlatformType
import org.springframework.stereotype.Component

/**
 * Factory for SocialDTO test instances.
 */
@Component
class SocialDTOFactory : BaseDtoFactory<SocialDTO>() {

    override fun targetType(): Class<SocialDTO> = SocialDTO::class.java

    override fun createBasic(): SocialDTO {
        val dto = SocialDTO()
        dto.title = "Hello world"
        dto.text = "Body"
        dto.url = "https://example.com/${nextId()}"
        dto.platforms = arrayOf(PlatformType.FACEBOOK)
        return dto
    }
}
